app/routes/next/project/$slug/detail.tsx

```diff
@@ -24,11 +54,75 @@ export const loader = async (args: DataFunctionArgs) => {
     }
   }
 
-  return json({ username }, { headers: response.headers });
+  const mode = await deriveProjectMode(sessionUser, slug);
+
+  const project = await prismaClient.project.findUnique({
+    select: {
+      slug: true,
+      name: true,
+      excerpt: true,
+      logo: true,
+      published: true,
+      background: true,
+    },
+    where: {
+      slug,
+    },
+  });
+  invariantResponse(project !== null, "Project not found", { status: 404 });
+
+  invariantResponse(
+    project.published || mode === "admin",
+    "This project isn't published yet.",
+    { status: 403 }
+  );
+
+  let background;
+  let blurredBackground;
+  if (project.background !== null) {
+    const publicURL = getPublicURL(authClient, project.background);
+    if (publicURL) {
+      background = getImageURL(publicURL, {
+        resize: { type: "fill", width: 1488, height: 480 },
+      });
+    }
+    blurredBackground = getImageURL(publicURL, {
+      resize: { type: "fill", width: 149, height: 48 },
+      blur: 5,
+    });
+  }
+  let logo;
+  let blurredLogo;
+  if (project.logo !== null) {
+    const publicURL = getPublicURL(authClient, project.logo);
+    if (publicURL) {
+      logo = getImageURL(publicURL, {
+        resize: { type: "fill", width: 248, height: 248 },
+      });
+    }
+    blurredLogo = getImageURL(publicURL, {
+      resize: { type: "fill", width: 25, height: 25 },
+      blur: 5,
+    });
+  }
+
+  const enhancedProject = {
+    ...project,
+    background,
+    blurredBackground,
+    logo,
+    blurredLogo,
+  };
+
+  return json(
+    { username, project: enhancedProject, mode },
+    { headers: response.headers }
+  );
 };
 
 function ProjectDetail() {
```

Comment:
There are some errors and warnings thrown in Browser Console.
`"Expected server HTML to contain a matching <input> in <div>."`
`"Hydration failed because the initial UI does not match what was rendered on the server."`
`"There was an error while hydrating. Because the error happened outside of a Suspense boundary, the entire root will switch to client rendering."`
`"(in promise) TypeError: Cannot read properties of undefined (reading 'length')"`

Reply:
Wrote ticket